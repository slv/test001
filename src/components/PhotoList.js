import React from "react";
import PropTypes from "prop-types";
import { PhotoList, PhotoLoading } from "./PhotoList.theme";

export const Component = ({ photo_list, photo_loading }) => (
  <PhotoList>
    {photo_loading ? (
      <PhotoLoading />
    ) : (
      (photo_list || []).map(photo_item => (
        <img
          key={photo_item.id}
          alt={photo_item.title}
          src={photo_item.thumbnailUrl}
        />
      ))
    )}
  </PhotoList>
);

Component.displayName = "PhotoList";

Component.propTypes = {
  photo_list: PropTypes.array,
  photo_loading: PropTypes.bool
};

export default Component;
