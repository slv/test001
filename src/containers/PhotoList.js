import React from "react";
import { connect } from "react-redux";
import { createSelector } from "reselect";
import { getPhotoList, getPhotoLoading } from "../selectors/getPhoto";
import PhotoList from "../components/PhotoList";
import { apiGet } from "../actions/api";

class PhotoListContainer extends React.Component {
  componentDidMount() {
    this.props.apiGet({
      path: "/photos",
      data: {
        _limit: 20
      }
    });
  }

  render() {
    return <PhotoList {...this.props} />;
  }
}

const mapStateToProps = () =>
  createSelector(
    getPhotoList,
    getPhotoLoading,
    (photo_list, photo_loading) => {
      return {
        photo_list,
        photo_loading
      };
    }
  );

const mapDispatchToProps = {
  apiGet
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PhotoListContainer);
