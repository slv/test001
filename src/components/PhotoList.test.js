import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import PhotoList from "./PhotoList";

describe("PhotoList", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<PhotoList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("match snapshot", () => {
    const renderedValue = renderer.create(<PhotoList />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});
