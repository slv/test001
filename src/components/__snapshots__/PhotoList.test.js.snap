// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoList match snapshot 1`] = `
<div
  className="sc-bdVaJa foRpMy"
/>
`;
